### PR TITLE
Feature/logger to main

### PR DESCRIPTION
**Pull request by GilLifshits**

### CODE REVIEW

File: utils/agent_handler.py [hunk 1]
```diff
@@ -20,8 +21,21 @@ def __init__(self, experiment, airsim_manager):
         self.action_space = spaces.Discrete(experiment.ACTION_SPACE_SIZE)
         self.observation_space = spaces.Box(low=-np.inf, high=np.inf, shape=(experiment.INPUT_SIZE,), dtype=np.float32)
         self.state = None
+        self.np_random = None
         self.reset()
 
+
+    def seed(self, seed=None):
```
Comment (GilLifshits) [inline utils/agent_handler.py:28]: Consider checking during runtime, in another file, that the seed in gym/random/np is indeed 42.

File: utils/experiment/experiment_config.py [hunk 2]
```diff
@@ -68,3 +74,18 @@ class Experiment:
     def __post_init__(self):
         self.EXPERIMENT_PATH = f"experiments/{self.EXPERIMENT_DATE_TIME}_{self.EXPERIMENT_ID}"
         self.SAVE_MODEL_DIRECTORY = f"{self.EXPERIMENT_PATH}/trained_model"
+
+        # Load API token from external JSON file
+        try:
+            with open("utils/logger/token.json", "r") as f:
```
Comment (GilLifshits) [inline utils/experiment/experiment_config.py:84]: add explanation in notion for setting up logger here: https://www.notion.so/Logger-1686974da452802aa5eef9540212b402

File: utils/experiment/experiment_config.py [hunk 3]
```diff
@@ -68,3 +74,18 @@ class Experiment:
...
+                config = json.load(f)
+                api_token = config["api_token"]
+        except (FileNotFoundError, KeyError) as e:
+            raise RuntimeError("Failed to load API token from config.json") from e
+
+        self.logger = NeptuneLogger(
+            project_name="AS-DRL/DRL-Research",
+            api_token=api_token,
+            run_name=self.EXPERIMENT_ID,
+            tags=["experiment", "training"]
```
Comment (GilLifshits) [inline utils/experiment/experiment_config.py]: define tags in experiment (not in post init) so that we can later inject different values in the experiment init.

File: utils/training_loop.py [hunk 4]
```diff
@@ -47,12 +47,30 @@ def training_loop(experiment, env, agent, model):
             all_rewards.append(episode_sum_of_rewards)
             all_actions.append(actions_per_episode)
 
+            # Log episode metrics
```
Comment (GilLifshits) [inline utils/training_loop.py:59]: Move all code chunks to another file (maybe: logger_utils), that has functions for logging metrics (pass metrics), function for saving model, function for hyperparams. 
Training loop should mostly include training loop code and not massive code of logging.

**Review by GilLifshits — COMMENTED** [review]
Great work
